Flaky Test: SeekableStreamSupervisorStateTest.testEmitRecordLag

`SeekableStreamSupervisorStateTest.testEmitRecordLag` failed in the ARM64 test pass. The [PR in which it failed](https://app.travis-ci.com/github/apache/druid/jobs/573408425) modified nothing in the supervisor area. The same test passes in the other architectures. This same PR passed this test in an earlier build. The change that triggered this build was tiny and in an area of code which the Supervisor does not use. Conclusion: something is flaky.

```text
[ERROR] org.apache.druid.indexing.seekablestream.supervisor.SeekableStreamSupervisorStateTest.testEmitBothLag  Time elapsed: 2.014 s  <<< ERROR!
java.util.ConcurrentModificationException
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1661)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.apache.druid.indexing.seekablestream.supervisor.SeekableStreamSupervisorStateTest.filterMetrics(SeekableStreamSupervisorStateTest.java:837)
	at org.apache.druid.indexing.seekablestream.supervisor.SeekableStreamSupervisorStateTest.testEmitBothLag(SeekableStreamSupervisorStateTest.java:651)
```

--- a/indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisorStateTest.java
+++ b/indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisorStateTest.java
@@ -24,6 +24,7 @@ import com.google.common.base.Optional;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.ImmutableSet;
+import com.google.errorprone.annotations.concurrent.GuardedBy;
 import org.apache.druid.data.input.impl.ByteEntity;
 import org.apache.druid.data.input.impl.DimensionSchema;
 import org.apache.druid.data.input.impl.DimensionsSpec;
@@ -1371,17 +1372,22 @@ public class SeekableStreamSupervisorStateTest extends EasyMockSupport
 
   private static class TestEmitter extends NoopServiceEmitter
   {
+    @GuardedBy("events")
     private final List<Event> events = new ArrayList<>();
 
     @Override
     public void emit(Event event)
     {
-      events.add(event);
+      synchronized (events) {
+        events.add(event);
+      }
     }
 
     public List<Event> getEvents()
     {
-      return events;
+      synchronized (events) {
+        return ImmutableList.copyOf(events);
+      }
     }
   }
 }

